content/en/docs/hertz/tutorials/toolkit/template.md

```diff
@@ -0,0 +1,473 @@
+├── main.go                       // Start the entrance 
+├── router.go                     // User-defined route write location 
+└── router_gen.go                 // hz generated route registration call
+```
+
+## IDL
+
+```
+// hello.thrift
+namespace go hello.example
+
+struct HelloReq {
+    1: string Name (api.query="name");
+}
+
+struct HelloResp {
+    1: string RespBody;
+}
+
+
+service HelloService {
+    HelloResp HelloMethod(1: HelloReq request) (api.get="/hello");
+}
+```
+
+## Command
+
+```
+hz new --mod=github.com/hertz/hello --idl=./hertzDemo/hello.thrift --customize_layout=template/layout.yaml:template/data.json
+```
+
+## The meaning of the default layout template
+
+> Note: The following bodies are all go templates
+
+```
+layouts:
+  # The directory of the generated handler will only be generated if there are files in the directory 
+  - path: biz/handler/
+    delims:
+      - ""
+      - ""
+    body: ""
+  # The directory of the generated model will only be generated if there are files in the directory 
+  - path: biz/model/
+    delims:
+      - ""
+      - ""
+    body: ""
+  # project main file,
+  - path: main.go
+    delims:
+      - ""
+      - ""
+    body: "// Code generated by hertz generator.\n\npackage main\n\nimport (\n\t\"github.com/cloudwego/hertz/pkg/app/server\"\n)\n\nfunc
+    main() {\n\th := server.Default()\n\n\tregister(h)\n\th.Spin()\n}\n"
+  # go.mod file, need template rendering data {{.GoModule}} to generate 
+  - path: go.mod
+    delims:
+      - '{{'
+      - '}}'
+    body: "module {{.GoModule}}\n{{- if .UseApacheThrift}}\nreplace github.com/apache/thrift
+    => github.com/apache/thrift v0.13.0\n{{- end}}\n"
+  # .gitignore file
+  - path: .gitignore
+    delims:
+      - ""
+      - ""
+    body: "*.o\n*.a\n*.so\n_obj\n_test\n*.[568vq]\n[568vq].out\n*.cgo1.go\n*.cgo2.c\n_cgo_defun.c\n_cgo_gotypes.go\n_cgo_export.*\n_testmain.go\n*.exe\n*.exe~\n*.test\n*.prof\n*.rar\n*.zip\n*.gz\n*.psd\n*.bmd\n*.cfg\n*.pptx\n*.log\n*nohup.out\n*settings.pyc\n*.sublime-project\n*.sublime-workspace\n!.gitkeep\n.DS_Store\n/.idea\n/.vscode\n/output\n*.local.yml\ndumped_hertz_remote_config.json\n\t\t
+    \ "
+  # .hz file, containing hz version, is the logo of the project created by hz, no need to transfer rendering data
+  - path: .hz
+    delims:
+      - '{{'
+      - '}}'
+    body: "
+      // Code generated by hz. DO NOT EDIT.
+
+      hz version: v{{.hzVersion}}"
+  # ping comes with ping handler 
+  - path: biz/handler/ping.go
+    delims:
+      - ""
+      - ""
+    body: "// Code generated by hertz generator.\n\npackage handler\n\nimport (\n\t\"context\"\n\n\t\"github.com/cloudwego/hertz/pkg/app\"\n\t\"github.com/cloudwego/hertz/pkg/common/utils\"\n)\n\n//
+    Ping .\nfunc Ping(ctx context.Context, c *app.RequestContext) {\n\tc.JSON(200,
+    utils.H{\n\t\t\"message\": \"pong\",\n\t})\n}\n"
+  # Define the file for route registration, need template rendering data {{.RouterPkgPath}} to generate 
```

Comment:
router_gen.go is the file that defines the route registration

Reply:
done